### ISSUE TITLE
Cannot Import from this file - Firefox 54.0.1 Password Exporter 1.3.4

### ISSUE THREAD

**Issue by doggyofone** [fligtar/password-exporter#79]
I had a problem with Firefox and had all of my passwords wiped out. I have a copy of my passwords in an excel xlsx format, which was converted from an exported csv file created through Password Exporter. I tried to save the xlsx file back into csv and after doing so I could not get the passwords imported by Password Exporter. I have tried copying the data  from the xlsx file into a blank csv spreadsheet and this will not import. I created a new password and exported it in csv format and then tried to copy all my passwords into that spreadsheet and I get the error "Cannot Import from this file". 
I'm sure I'm missing something simple, but need some help likely with the format of the csv sheet. In using Excel the conversion to csv is based on CSV (MS DOS), with no other option available (UTF8). 
Thanks for your anticipated help.

**Comment by JustOff** [comment]
> I created a new password and exported it in csv format

Open this csv with a text editor, it will have 2-line header. You should always preserve it or import will fail.

**Comment by jvillalobos** [comment]
Yes, the first two lines are meant to ensure that the CSV was generated by the Password Exporter extension, so they are required for the import.

If you still have trouble with this, please post back. I'm closing this now because there's no issue to fix in the extension.

**Comment by doggyofone** [comment]
Thanks for the help/ Where can I find the csv you mention above?

**Comment by jvillalobos** [comment]
The file you exported in CSV format using the extension should have those two lines. Just open it in a text editor and you should see them.

You can also try with this sample file: [sample.txt](https://github.com/fligtar/password-exporter/files/1133374/sample.txt). I changed the extension to txt because GitHub doesn't support CSV.

**Comment by doggyofone** [comment]
OK, I created a csv file with 6 recent passwords added since the loss of all my passwords a couple days ago. I opened it and added the data from the xlxs file in the appropriate columns, saved and then opened password exporter. The imported the csv, but the Import/export window all the buttons grey out. The import window that shows progress is empty with the words cancel import shown just below. No import occurs. Which text editor would you recommend for Windows 10?

**Comment by jvillalobos** [comment]
A simple text editor like Notepad should work, I think. I'm not familiar with what's available on Windows 10.

**Comment by doggyofone** [comment]
Could you email the sample.txt. I cannot open it, or change the extension.

**Comment by doggyofone** [comment]
OK, I saved my recent passwords as csv and opened the csv sheet. The two lines row 1&2 are as follows:

1
--
# Generated by Password Exporter; Export format 1.1; Encrypted: false
hostname | username | password | formSubmitURL | httpRealm | usernameField | passwordField
I copied the data from my xlsx spreadsheet into the correct columns below these two lines. When I try to import the data via Password Exporter the buttons go grey and no import starts. 
I'r really stuck here. Is there anything I'm missing or I should try?

**Comment by JustOff** [comment]
> I saved my recent passwords as csv and opened the csv sheet. The two lines row 1&2 are as follows

No, they should look different. Do not use anything other than a text editor (eg [Notepad++](https://notepad-plus-plus.org/)).

**Comment by doggyofone** [comment]
The problem is that I don't know what they should look like. If I did I would use the editor to make the changes. 
The copy/paste I did in the note above doesn't show exactly how the spreadsheet looks. 
This line - # Generated by Password Exporter; Export format 1.1; Encrypted: false - is found in cell A1
These values are found in cells A2 through G2 when I copy and paste them:
1 | 2 | 3 | 4 | 5 | 6 | 7
-- | -- | -- | -- | -- | -- | --
hostname | username | password | formSubmitURL | httpRealm | usernameField | passwordField
I'm not sure where the first two lines come from, as they appeared when I pasted the title line. 
Please advise. I have close to 300 passwords that I am trying to rescue.



![csv pic](https://user-images.githubusercontent.com/30008544/27994222-073c4a70-6487-11e7-9fb1-de8fe52da26f.JPG)

**Comment by JustOff** [comment]
@doggyofone 
**Do not use anything other than a text editor!**

**Comment by doggyofone** [comment]
I understand. I have Notepad++. When I open the xlsx spreadsheet it opens into a mess of data. I saved the xlxs as a csv file and opened it with Notepad++ and it looks fine. I saved it as a csv in Notepad ii, opened it and it looks OK. Tried importing and nothing happened.
The question is what am I doing to the csv file with the editor. I don't know what the spreadsheet should look like, or why it won't import.

**Comment by JustOff** [comment]
You have to ensure that csv file is proper, ie:
* 2-lines header exists
* every line has all the fields
* each field is enclosed in quotes

**Comment by doggyofone** [comment]
I opened the csv in Notepad ii. I added quotes to each column title and saved as csv. When I open the spreadsheet no quotations have been saved. Note that I am saving it as a .csv spreadsheet All types (*.*) as sav as type - no csv option in notepad ii.
This is what Notepad looks like:
![notepad jpg](https://user-images.githubusercontent.com/30008544/27994525-d8649ffc-648d-11e7-9a3c-659c051decc0.JPG)
This is what csv looks like:
![csv snip](https://user-images.githubusercontent.com/30008544/27994555-6731dc7c-648e-11e7-81a5-ea89a917e1ab.JPG)
Bottom line is that it does not import.

**Comment by JustOff** [comment]
I'm sorry, but I have nothing more to add, but unfortunately it seems you don't hear me. Try to read what I wrote above once again and for the last time: do not use anything but the text editor to prepare the final csv file for import.

**Comment by doggyofone** [comment]
It may just be the language differences, but your instructions aren't clear enough for me. 
I have tried about every combination I can think of and edited with notepad ii. Nothing imports.
If I could see a sample of what you are trying to explain it would help a lot. The sample.txt you sent above cannot be opened as anything other than xml and saving it as csv and opening in notepad ii does not work. It ends up with the xml mess on a csv. Please email the sample.txt.

**Comment by jvillalobos** [comment]
I sent you the sample file. Hope it helps. However, if that doesn't work, I suggest you look for someone who is more technical (no offense intended) to help you with this in person.
